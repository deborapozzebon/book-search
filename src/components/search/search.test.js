import { render, screen, fireEvent } from '@testing-library/react';
import Search from './search';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
  const div = document.createElement("div");
  ReactDOM.render(<Search ></Search>, div);
  ReactDOM.unmountComponentAtNode(div);
})

test('snapshot 1', () => {
  const book = [{
    title: 'JavaScript',
    Description: 'JavaScript book to begginers'
  }]
  const tree = renderer.create(<Search books={book} />).toJSON();
  expect(tree).toMatchSnapshot();
})

test('hadlerSubmit', () => {
  const handler = jest.fn(e => e.preventDefault())
  const { getByTestId } = render(
    <form onSubmit={handler} data-testid="form">
      <button type="submit">Submit</button>
    </form>,
  )
  fireEvent.submit(getByTestId('form'))
  expect(handler).toHaveBeenCalledTimes(1)
})