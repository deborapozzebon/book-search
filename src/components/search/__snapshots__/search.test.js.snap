// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group mt-2"
  >
    <input
      className="form-control mt-2"
      data-testid="search-input"
      onChange={[Function]}
      placeholder="Book name"
      type="text"
    />
    <button
      className="btn btn-primary mt-2"
      data-testid="search-button"
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;
